gobblin-api/src/main/java/org/apache/gobblin/util/io/GsonInterfaceAdapter.java

```diff
@@ -189,7 +192,38 @@ private <S> S readValue(JsonObject jsonObject, TypeToken<S> defaultTypeToken) th
   }
 
   public static <T> Gson getGson(Class<T> clazz) {
-    Gson gson = new GsonBuilder().registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz)).create();
+    Gson gson = new GsonBuilder()
+        .setObjectToNumberStrategy(CustomToNumberPolicy.INTEGER_OR_LONG_OR_DOUBLE)
+        .registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz))
+        .create();
     return gson;
   }
+
+  public enum CustomToNumberPolicy implements ToNumberStrategy {
+    INTEGER_OR_LONG_OR_DOUBLE {
+      @Override
+      public Number readNumber(JsonReader in) throws IOException {
+        String value = in.nextString();
+        try {
+          return Integer.parseInt(value);
+        } catch (NumberFormatException var3) {
```

Comment:
nit : a better way for naming exceptions in this scenario could be ignored 2/3, instead of var 2/3.

Reply:
+1 to this